src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2104,15 +2100,49 @@ class BeginEndSegment(BaseSegment):
         Ref("DelimiterSegment", optional=True),
         Indent,
         AnyNumberOf(
-            OneOf(
-                Ref("BeginEndSegment"),
+            Ref("StatementSegment"),
```

Comment:
Do we not also want a `Ref("DelimiterSegment", optional=True),` after this line, like we've added to TryCatchSegment?

Reply:
Yes, good catch!